File: src/test/java/Examples.java
import com.kylecorry.frc.vision.*;
import edu.wpi.cscore.UsbCamera;
import edu.wpi.first.wpilibj.CameraServer;
import org.junit.Test;

import java.util.List;

/**
 * Created by Kylec on 2/12/2017.
 */
public class Examples {

    public void testCameraSource(){
        UsbCamera usbCamera = CameraServer.getInstance().startAutomaticCapture(0);
        usbCamera.setExposureManual(0);
        usbCamera.setBrightness(0);
        usbCamera.setWhiteBalanceManual(10000);
        usbCamera.setResolution(160, 120);

        CameraSource cameraSource = new CameraSource(usbCamera);

        TargetGroupDetector detector = new TargetGroupDetector(new ExampleSpecs(), new ExampleGroupSpecs());

        List<TargetGroup> targets = detector.detect(cameraSource.getPicture());

        if(!targets.isEmpty()){
            TargetGroup bestTarget = targets.get(0);

            double angle = bestTarget.computeAngle(160, CameraSpecs.MicrosoftLifeCam.HORIZONTAL_VIEW_ANGLE);
            double distance = bestTarget.computeDistance(160, new ExampleGroupSpecs().getGroupWidth(), CameraSpecs.MicrosoftLifeCam.HORIZONTAL_VIEW_ANGLE);

            double centerX = bestTarget.getCenterPosition().x;

            double confidence = bestTarget.getIsTargetGroupProbability(); // 0 - 1

            if(confidence > 0.75){
                // do something
            }
        }

    }


}

File: src/test/java/Examples.kt
import com.kylecorry.frc.vision.camera.CameraSource
import com.kylecorry.frc.vision.camera.CameraSpecs
import com.kylecorry.frc.vision.detection.TargetGroup
import com.kylecorry.frc.vision.detection.TargetGroupDetector
import edu.wpi.cscore.UsbCamera
import edu.wpi.first.wpilibj.CameraServer

/**
 * Created by Kylec on 2/12/2017.
 */
class Examples {

    fun testCameraSource() {
        val usbCamera = CameraServer.getInstance().startAutomaticCapture(0)
        usbCamera.setExposureManual(0)
        usbCamera.brightness = 0
        usbCamera.setWhiteBalanceManual(10000)
        usbCamera.setResolution(160, 120)

        val cameraSource = CameraSource(usbCamera)

        val detector = TargetGroupDetector(ExampleSpecs(), ExampleGroupSpecs())

        val targets = detector.detect(cameraSource.picture)

        if (!targets.isEmpty()) {
            val bestTarget = targets[0]

            val angle = bestTarget.computeAngle(CameraSpecs.MicrosoftLifeCam.HORIZONTAL_VIEW_ANGLE)
            val distance = bestTarget.computeDistance(ExampleGroupSpecs().groupWidth, CameraSpecs.MicrosoftLifeCam.HORIZONTAL_VIEW_ANGLE)

            val centerX = bestTarget.centerPosition.x

            val confidence = bestTarget.probability // 0 - 1

            if (confidence > 0.75) {
                // do something
            }
        }

    }


}
